scale_test/dnspolicy/namespaced-dns-operator-deployments.yaml

```diff
@@ -0,0 +1,162 @@
+        kind: Secret
+        replicas: 1
+        inputVars:
+          KUADRANT_GCP_GOOGLE_CREDENTIALS: '{{ .KUADRANT_GCP_GOOGLE_CREDENTIALS }}'
+          KUADRANT_GCP_PROJECT_ID: '{{ .KUADRANT_GCP_PROJECT_ID }}'
+      {{end}}
+      {{if (eq .DNS_PROVIDER "azure")}}
+      - objectTemplate: https://raw.githubusercontent.com/{{.DNS_OPERATOR_GITHUB_ORG}}/dns-operator/refs/heads/{{.DNS_OPERATOR_GITREF}}/test/scale/kubeburner-object-templates/azure-credentials.yaml
+        kind: Secret
+        replicas: 1
+        inputVars:
+          KUADRANT_AZURE_CREDENTIALS: '{{ .KUADRANT_AZURE_CREDENTIALS }}'
+      {{end}}
+  - name: dnspolicy-scale-test-loadbalanced
+    jobIterations: {{ .JOB_ITERATIONS }}
+    namespacedIterations: true
+    namespace: scale-test
+    waitWhenFinished: true
+    maxWaitTimeout: 2m
+    preLoadImages: false
+    objects:
+    {{- $numGWs := .NUM_GWS | atoi }}
+    {{- range $index := until $numGWs }}
+      {{- $gwNum := add1 $index }}
+      - objectTemplate: gw.yaml
+        replicas: 1
+        waitOptions:
+          forCondition: Programmed
+          customStatusPath: ".conditions[].type"
+        inputVars:
+          KUADRANT_ZONE_ROOT_DOMAIN: '{{ $.KUADRANT_ZONE_ROOT_DOMAIN }}'
+          NUM_LISTENERS: '{{ $.NUM_LISTENERS }}'
+          GW_NUM: '{{ $gwNum }}'
+      - objectTemplate: gw-dnspolicy-loadbalanced.yaml
+        kind: DNSPolicy
+        replicas: 1
+        inputVars:
+          KUADRANT_ZONE_ROOT_DOMAIN: '{{ $.KUADRANT_ZONE_ROOT_DOMAIN }}'
+          GW_NUM: '{{ $gwNum }}'
+          DNS_PROVIDER: '{{ $.DNS_PROVIDER }}'
+          GeoCode: {{if (eq $.DNS_PROVIDER "aws")}} GEO-EU {{else if (eq $.DNS_PROVIDER "gcp")}} europe-west1 {{else if (eq $.DNS_PROVIDER "azure")}} GEO-EU {{else}} foo {{end}}
+        waitOptions:
+          forCondition: "Enforced"
+          customStatusPath: ".conditions[].type"
+      {{- range $index := until ($.NUM_LISTENERS | atoi) }}
+      {{- $listenerNum := add1 $index }}
+      - objectTemplate: ./httproute.yaml
+        replicas: 1
+        waitOptions:
+          forCondition: Accepted
+          customStatusPath: ".conditions[].type"
+        inputVars:
+          KUADRANT_ZONE_ROOT_DOMAIN: '{{ $.KUADRANT_ZONE_ROOT_DOMAIN }}'
+          GW_NUM: '{{ $gwNum }}'
+          LISTENER_NUM: "{{$listenerNum}}"
+      {{- end }}
+    {{- end }}
+  - name: scale-test-dnspolicy-read
+    jobType: read
+    jobIterations: 1
+    jobPause: 30s
+    objects:
+      - kind: DNSPolicy
+        apiVersion: kuadrant.io/v1
+        labelSelector: {kube-burner-job: dnspolicy-scale-test-loadbalanced}
+  {{ if (eq .SKIP_CLEANUP "true") }}
+  # nothing to do if cleanup is skipped
+  {{ else }}
+  - name: dnspolicy-scale-test-cleanup
+    jobType: delete
+    objects:
+      - kind: DNSPolicy
+        apiVersion: kuadrant.io/v1alpha1
+        labelSelector: {kube-burner-job: dnspolicy-scale-test-loadbalanced}
+  {{ end }}
```

Comment:
I copied the pattern from the [original scale test](https://github.com/Kuadrant/testsuite/blob/main/scale_test/config.yaml#L139) of adding this cleanup to remove the DNSPolicies. I understand why its here, but it is fairly frustrating that we need it.

I wonder if we should revisit the cleanup of policies/records/secrets and see if there is some reasonable way we can prevent secrets being removed before all resources referencing them are deleted.

@maleck13 I know we have discussed this before, but i think its probably the single most annoying thing about testing anything DNS related.

Reply:
Yes. Well I guess the only real option is to add a finalizer to secret once referenced by a policy as a provider. I guess as the perm there would just be to add fnalizers it wouldn't mean we have full write access to secrets. The more I think about the more reasonable it seems to add a finalizer and remove it if the ref changes. As more than one policy can reference a single secret, we would likely need some form of ref count adding as an annotation for example? 
Anyway I think it makes sense to add a new issue around this given how much hassle it causes people

Reply:
Issue to look into this more here https://github.com/Kuadrant/dns-operator/issues/344